trading_calendars/exchange_calendar_xses.py

```diff
@@ -0,0 +1,374 @@
+    "2005-02-10",
+    "2005-03-25",
+    "2005-05-02",
+    "2005-05-23",
+    "2005-08-09",
+    "2005-11-01",
+    "2005-11-03",
+    "2005-12-26",
+    "2006-01-02",
+    "2006-01-10",
+    "2006-01-30",
+    "2006-01-31",
+    "2006-04-14",
+    "2006-05-01",
+    "2006-05-12",
+    "2006-08-09",
+    "2006-10-24",
+    "2006-12-25",
+    "2007-01-01",
+    "2007-01-02",
+    "2007-02-19",
+    "2007-02-20",
+    "2007-04-06",
+    "2007-05-01",
+    "2007-05-31",
+    "2007-08-09",
+    "2007-11-08",
+    "2007-12-20",
+    "2007-12-25",
+    "2008-01-01",
+    "2008-02-07",
+    "2008-02-08",
+    "2008-03-21",
+    "2008-05-01",
+    "2008-05-19",
+    "2008-10-01",
+    "2008-10-27",
+    "2008-12-08",
+    "2008-12-25",
+    "2009-01-01",
+    "2009-01-26",
+    "2009-01-27",
+    "2009-04-10",
+    "2009-05-01",
+    "2009-08-10",
+    "2009-09-21",
+    "2009-11-27",
+    "2009-12-25",
+    "2010-01-01",
+    "2010-02-15",
+    "2010-02-16",
+    "2010-04-02",
+    "2010-05-28",
+    "2010-08-09",
+    "2010-09-10",
+    "2010-11-05",
+    "2010-11-17",
+    "2011-02-03",
+    "2011-02-04",
+    "2011-04-22",
+    "2011-05-02",
+    "2011-05-17",
+    "2011-08-09",
+    "2011-08-30",
+    "2011-10-26",
+    "2011-11-07",
+    "2011-12-26",
+    "2012-01-02",
+    "2012-01-23",
+    "2012-01-24",
+    "2012-04-06",
+    "2012-05-01",
+    "2012-08-09",
+    "2012-08-20",
+    "2012-10-26",
+    "2012-11-13",
+    "2012-12-25",
+    "2013-01-01",
+    "2013-02-11",
+    "2013-02-12",
+    "2013-03-29",
+    "2013-05-01",
+    "2013-05-24",
+    "2013-08-08",
+    "2013-08-09",
+    "2013-10-15",
+    "2013-12-25",
+    "2014-01-01",
+    "2014-01-31",
+    "2014-04-18",
+    "2014-05-01",
+    "2014-05-13",
+    "2014-07-28",
+    "2014-10-06",
+    "2014-10-22",
+    "2014-12-25",
+    "2015-01-01",
+    "2015-02-19",
+    "2015-02-20",
+    "2015-04-03",
+    "2015-05-01",
+    "2015-06-01",
+    "2015-07-17",
+    "2015-08-07",
+    "2015-08-10",
+    "2015-09-11",
+    "2015-09-24",
+    "2015-11-10",
+    "2015-12-25",
+    "2016-01-01",
+    "2016-02-08",
+    "2016-02-09",
+    "2016-03-25",
+    "2016-05-02",
+    "2016-07-06",
+    "2016-08-09",
+    "2016-09-12",
+    "2016-12-26",
+    "2017-01-02",
+    "2017-01-30",
+    "2017-04-14",
+    "2017-05-01",
+    "2017-05-10",
+    "2017-06-26",
+    "2017-08-09",
+    "2017-09-01",
+    "2017-10-18",
+    "2017-12-25",
+    "2018-01-01",
+    "2018-02-16",
+    "2018-03-30",
+    "2018-05-01",
+    "2018-05-29",
+    "2018-06-15",
+    "2018-08-09",
+    "2018-08-22",
+    "2018-11-06",
+    "2018-12-25",
+    "2019-01-01",
+    "2019-02-05",
+    "2019-02-06",
+    "2019-04-19",
+    "2019-05-01",
+    "2019-06-05",
+    "2019-08-09",
+    "2019-12-15",
+    "2020-01-01",
+    "2020-01-27",
+    "2020-04-10",
+    "2020-05-01",
+    "2020-05-07",
+    "2020-05-24",
+    "2020-07-31",
+    "2020-12-25",
+])
+
+
+class XSESExchangeCalendar(PrecomputedTradingCalendar):
```

Comment:
According to @jmccorriston Singapore has half days on: the day prior to Chinese New Year, Christmas Eve, and New Year's Eve. The definitions for these are missing in this calendar.

Reply:
Good catch. Going to skip now and add TODO, because half days don't affect us until we have minutely simulation.